apps/settings/test/unit/mock_dialog_service.js

```diff
@@ -0,0 +1,11 @@
+define(function() {
```

Comment:
Thanks for adding this mock. Would you be able to remove the existing ones (that recreate it in other tests?):
* test/unit/panels/addon_details/panel_test.js
* test/unit/panels/call_fdn_list/panel_test.js
* test/unit/panels/call_fdn_settings/panel_test.js
* test/unit/panels/developer/developer_test.js
* test/unit/panels/full_developer_mode/panel_test.js
* test/unit/panels/screen_lock/screen_lock_test.js
* test/unit/panels/simpin/simpin_test.js
* test/unit/panels/simpin_dialog/simpin_dialog_test.js
* test/unit/panels/usb_storage/usb_transfer_test.js
No worries if you don't have cycles for that.